homeassistant/components/snmp/switch.py

```diff
@@ -191,17 +211,45 @@ def __init__(
 
     async def async_turn_on(self, **kwargs):
         """Turn on the switch."""
-        if self._command_payload_on.isdigit():
-            await self._set(Integer(self._command_payload_on))
-        else:
-            await self._set(self._command_payload_on)
+        # If vartype set, use it - http://snmplabs.com/pysnmp/docs/api-reference.html#pysnmp.smi.rfc1902.ObjectType
+        await self._execute_command(self._command_payload_on)
 
     async def async_turn_off(self, **kwargs):
         """Turn off the switch."""
-        if self._command_payload_on.isdigit():
-            await self._set(Integer(self._command_payload_off))
+        await self._execute_command(self._command_payload_off)
+
+    async def _execute_command(self, command):
+        if self._vartype == "Null":
+            await self._set(Null(""))
```

Comment:
Can we make a dict that maps vartype to the type class, then look up that dict here and if the result is not None, we set the instance?

This will just leave three special cases it looks like, Null type being one.

Reply:
Updated based on your comments, not sure if the dict would fit in const.py instead of switch.py, but probably not. I kept it in switch.py. 

Documentation: https://github.com/home-assistant/home-assistant.io/pull/13249